src/app/clients/page.tsx

```diff
@@ -11,14 +16,33 @@ const dataFetch: () => Promise<Schema["clients"][] | null> = async () => {
 const Clients: () => Promise<React.ReactElement> = async () => {
     const data = await dataFetch();
 
+    if (data === null) {
+        <>
+            <h1> Clients Page </h1>
+            <p> Error: Data not found</p>
+        </>;
+    }
+
+    const headers: { [key: string]: string } = {
+        full_name: "Name",
+        phone_number: "Phone Number",
+        address_1: "Address",
+    };
+
+    const dataWithoutKey: DataForTable[] = data!.map((datum: Schema["clients"]) => {
+        return {
+            full_name: datum.full_name,
+            phone_number: datum.phone_number,
+            address_1: datum.address_1,
+        };
+    });
     return (
         <>
             <h1> Clients Page </h1>
 
-            <p> Testing Supabase fetching </p>
-
             {/* This should be a separate component which is passed data via props */}
-            <pre>{JSON.stringify(data, null, 4)}</pre>
+            {/* <pre>{JSON.stringify(data, null, 4)}</pre> */}
+            <Table data={dataWithoutKey} headers={headers} />
         </>
```

Comment:
Should wrap the logic for rendering the `Table `component in an else block if `data `is `null`. Currently will render both, even if `data` is `null`.

Reply:
We have implemented the logic